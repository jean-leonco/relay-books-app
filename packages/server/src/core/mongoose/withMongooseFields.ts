export const isActiveMongooseField = {
  isActive: {
    type: Boolean,
    description: 'Soft delete. Usually when users want to hide something from public view.',
    required: true,
    default: true,
    index: true,
  },
};

export const removedAtMongooseField = {
  removedAt: {
    type: Date,
    description: 'Hard delete',
    default: null,
  },
};
