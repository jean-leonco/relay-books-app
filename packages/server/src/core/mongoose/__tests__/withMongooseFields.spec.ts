import * as withMongooseFields from '../withMongooseFields';

it('should snapshot mongoose fields', async () => {
  expect({ ...withMongooseFields }).toMatchSnapshot();
});
