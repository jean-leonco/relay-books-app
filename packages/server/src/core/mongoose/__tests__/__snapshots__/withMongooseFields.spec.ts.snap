// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot mongoose fields 1`] = `
Object {
  "isActiveMongooseField": Object {
    "isActive": Object {
      "default": true,
      "description": "Soft delete. Usually when users want to hide something from public view.",
      "index": true,
      "required": true,
      "type": [Function],
    },
  },
  "removedAtMongooseField": Object {
    "removedAt": Object {
      "default": null,
      "description": "Hard delete",
      "type": [Function],
    },
  },
}
`;
